pkg/synopsysctl/cmd_destroy.go

```diff
@@ -89,28 +95,35 @@ func destroy(namespace string) error {
 		if crdNames, ok := cmData["CrdNames"]; ok {
 			crds = strings.Split(crdNames.(string), ",")
 		}
+		isExist = true
 	}
 
 	// delete namespace
-	checkErr := util.CheckResourceNamespace(restconfig, kubeClient, strings.Join(crds, ","), namespace, true)
-	if err != nil && isForceDestroy {
-		log.Warnf("%s. namespace cannot be deleted", err.Error())
-	} else if err != nil && !isForceDestroy {
-		return fmt.Errorf("%s. It is not recommended to destroy the Synopsys Operator so these resources can continue to be managed. If you are sure you want to delete the Synopsys Operator anyway then you can use the 'force' option which will keep all the instances and delete only the Synopsys Operator", err.Error())
-	}
+	if isExist {
+		isExist, err := util.CheckResourceNamespace(restconfig, kubeClient, strings.Join(crds, ","), namespace, true)
+		if isExist && err != nil && isForceDestroy {
+			log.Warnf("%s. namespace cannot be deleted", err.Error())
+		} else if isExist && err != nil && !isForceDestroy {
+			return fmt.Errorf("%s. It is not recommended to destroy the Synopsys Operator so these resources can continue to be managed. If you are sure you want to delete the Synopsys Operator anyway then you can use the 'force' option which will keep all the instances and delete only the Synopsys Operator", err.Error())
+		} else if err != nil {
+			log.Error(err)
+		}
 
-	log.Infof("deleting the Synopsys Operator in namespace '%s'", namespace)
-	// delete Synopsys Operator instance
-	commonConfig := crdupdater.NewCRUDComponents(restconfig, kubeClient, false, false, namespace, &api.ComponentList{}, "app=synopsys-operator", false)
-	_, crudErrors := commonConfig.CRUDComponents()
-	if len(crudErrors) > 0 {
-		log.Errorf("unable to delete the Synopsys Operator in namespace '%s' due to %+v", namespace, err)
-	}
+		// delete Synopsys Operator instance
+		if isExist {
```

Comment:
Why this check again?  Doesn't `ifExist` already have to be true to even get inside this conditional (i.e.: line 102)?

Reply:
nm, i see it's reset in line 103, maybe control flow readability could be improved

Reply:
done